core/src/main/java/software/amazon/awssdk/core/http/DefaultErrorResponseHandler.java

```diff
@@ -89,11 +89,11 @@ private AmazonServiceException createAse(HttpResponse errorResponse) throws Exce
          * unmarshall the response, but we might need something a little more
          * sophisticated in the future.
          */
-        for (Unmarshaller<AmazonServiceException, Node> unmarshaller : unmarshallerList) {
-            AmazonServiceException ase = unmarshaller.unmarshall(document);
-            if (ase != null) {
-                ase.setStatusCode(errorResponse.getStatusCode());
-                return ase;
+        for (Unmarshaller<SdkServiceException, Node> unmarshaller : unmarshallerList) {
+            SdkServiceException exception = unmarshaller.unmarshall(document);
+            if (exception != null) {
+                exception.statusCode(errorResponse.getStatusCode());
```

Comment:
could `errorResponse` be null?

Reply:
I'm not currently aware of a real case where it could be null.

Reply:
it's not massively expensive nor that big a maintenance burden - prolly just leave it in for now.